### PR TITLE
[CI] Fix CI to only use https for git installs and dev.py update

### PR DESCRIPTION
**Pull request by adam2392**
Fixes #

Changes proposed in this pull request:

-

## Before submitting

- [ ] I've read and followed all steps in the [Making a pull request](https://github.com/py-why/pywhy-graphs/blob/main/CONTRIBUTING.md#making-a-pull-request)
    section of the `CONTRIBUTING` docs.
- [ ] I've updated or added any relevant docstrings following the syntax described in the
    [Writing docstrings](https://github.com/py-why/pywhy-graphs/blob/main/CONTRIBUTING.md#writing-docstrings) section of the `CONTRIBUTING` docs.
- [ ] If this PR fixes a bug, I've added a test that will fail without my fix.
- [ ] If this PR adds a new feature, I've added tests that sufficiently cover my new functionality.

## After submitting

- [ ] All GitHub Actions jobs for my pull request have passed.

### CODE REVIEW

**Comment by adam2392** [comment]
The CI now works as it should. Please review the changes, and I will let you merge it in @jshinm.

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -49,10 +50,10 @@ jobs:
...
 
       # Make sure the @v0.7.0 matches the current version of the action
-      - name: webfactory/ssh-agent
-        uses: webfactory/ssh-agent@v0.7.0
-        with:
-          ssh-private-key: ${{ secrets.SSH_PRIVATE_KEY }}
+      # - name: webfactory/ssh-agent
+      #   uses: webfactory/ssh-agent@v0.7.0
+      #   with:
+      #     ssh-private-key: ${{ secrets.SSH_PRIVATE_KEY }}
```
Comment (adam2392) [inline .github/workflows/main.yml]:
```suggestion
```

I suppose we can remove these lines

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -246,9 +155,9 @@ jobs:
 
       # makes sure that SSH key allows you to pull from another repository
       # Make sure the @v0.7.0 matches the current version of the action
```
Comment (adam2392) [inline .github/workflows/main.yml]:
```suggestion
```

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -246,9 +155,9 @@ jobs:
...
-      - uses: webfactory/ssh-agent@v0.7.0
-        with:
-          ssh-private-key: ${{ secrets.SSH_PRIVATE_KEY }}
+      # - uses: webfactory/ssh-agent@v0.7.0
+      #   with:
+      #     ssh-private-key: ${{ secrets.SSH_PRIVATE_KEY }}
```
Comment (adam2392) [inline .github/workflows/main.yml]:
```suggestion
```

File: .github/workflows/main.yml [hunk 4]
```diff
@@ -49,10 +50,10 @@ jobs:
           cache-dependency-path: "requirements.txt"
 
       # Make sure the @v0.7.0 matches the current version of the action
```
Comment (adam2392) [inline .github/workflows/main.yml]:
```suggestion
```

**Review by jshinm — APPROVED** [review]

**Pull request merged**